contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,571 @@
+with the same label.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+**volumeType: block**
+accessModes:
+- ReadWriteOnce
+resources:
+requests:
+storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the user also specifically defines the intent of the volume by 
+indicating the volumeType as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+**storageClassName: local-fast**
+**volumeType: block**
+accessModes:
+- ReadWriteOnce
+resources:
+requests:
+storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeType will default to volumes work today, which are formatted with a filesystem depending on 
+the plug-in chosen. Recycling will not be a supported reclaim policy. Once the user deletes the claim against a PV, the volume will 
+be scrubbed according to how it was bound. The path value in the PV definition would be overloaded to define the path of the raw
+block device rather than the fileystem path.
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+**volumeType: block**
+capacity:
+storage: 100Gi
+local:
+**path: /dev/xvdc**
+accessModes:
+  - ReadWriteOnce
+persistentVolumeReclaimPolicy: Delete 
+```
+
+## Storage Class API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeType
+as 'block' but the fsType as 'xfs'.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: kubernetes.io/local-block-glusterfs
+parameters:
+**volumeType: block**
+fsType: xfs
+```
+The provisioner should validate the parameters and return and error if the combination specified is not supported.
+This also allows the use case for leveraging a Storage Class for utilizing pre-defined static volumes.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: no-provisioning 
+parameters:
+**volumeType: block**
+```
+  
+# Use Cases
+
+## UC1: 
+
+DESCRIPTION: 
+
+A developer wishes to enable their application to use a local raw block device as the volume for the container. The admin has already created PVs that the user will bind to by specifying 'block' as the volume type of their PVC.
```

Comment:
Currently, most of storage plugins are network attached, I think we should introduce examples mainly base on network attached storage plugin such as gce, aws, fc, iscsi, nfs etc, and then we can add local storage plugin(local storage case) as additional examples, but we need clearly distinguish between network attached case and local storage case to avoid confusion.